test.sh

```diff
@@ -6,8 +6,37 @@
 Me=$(basename "$0")
 set -euo pipefail
 
-# location of binaries
-BINDIR="${BINDIR:-bin}"
+# location of binaries, which live under $BUILD_DIR, if set.
```

Comment:
This whole block was needed just to allow someone to run `test.sh` by itself.

One use-case is to export these `BUILD_` env-vars in my session, do the `make` and then do the `test.sh` to run tests.

I don't recall fully now, but it _could be the case_ that CI-jobs may run even w/o this fix. Certainly, manual invocation of `test.sh` w/o specifying `BINDIR` on the env will certainly fail.

Reply:
Btw, I recall now ... In all CI-scripts, the invocation is `make all run-tests`, with an earlier change (See PR #373).

Prior to that commit of 0edbba6, the CI-scripts used to execute the tests as: `make && ./test.sh`.

With that usage, this chunk of code in `test.sh` to establish `BINDIR` was necessary.

With the rework to run tests using `make all run-tests`, we really don't need this chunk as part of this change set, but to not regress other dev's usage, we should include this chunk in this change-set.